lib/puppet-strings/hiera/hierarchy_data_path.rb

```diff
@@ -13,20 +13,17 @@ def initialize(datadir, path)
     def matches
       result = {}
 
-      Dir.chdir(datadir) do
-        Dir['**'].each do |entry|
-          next unless File.file?(entry)
+      Dir.glob(File.join(datadir, '**', '*.yaml')).each do |entry|
+        next unless File.file?(entry)
 
-          regex.match(entry) do |match|
-            full_path = File.join(datadir, entry)
-            interpolations = {}
+        regex.match(entry) do |match|
```

Comment:
hmm im not sure thats the best idea.  1) im not convinced there is a problem to solve (do you have some examples, even contrived ones) 2) if there is a problem at best we are just reducing the risk of triggering it by using chdir

Reply:
The alternative is to normalize it again. For example (using [pathname](https://ruby-doc.org/stdlib-2.6.3/libdoc/pathname/rdoc/Pathname.html)):
```ruby
data_dir = Pathname.new(datadir)
data_dir.glob('**').each do |entry|
  next unless x.file?

  regex.match(entry.relative_path_from(data_dir)) do |match|
    # ...
  end
end
```

Note that I also chose to glob everything so that we don't have to think about extensions (yaml vs yml). The regex matching will take care of that.